planner_cspace/src/planner_3d.cpp

```diff
@@ -394,6 +396,124 @@ class Planner3dNode
 
     return true;
   }
+  bool cbMakePlan(nav_msgs::GetPlan::Request &req,
+                  nav_msgs::GetPlan::Response &res)
+  {
+    if (!has_map_)
+      return false;
+
+    Astar::Vec s, e;
+    metric2Grid(s[0], s[1], s[2],
+                req.start.pose.position.x, req.start.pose.position.y, tf::getYaw(req.start.pose.orientation));
+    s.cycleUnsigned(s[2], map_info_.angle);
+    metric2Grid(e[0], e[1], e[2],
+                req.goal.pose.position.x, req.goal.pose.position.y, tf::getYaw(req.goal.pose.orientation));
+    e.cycleUnsigned(e[2], map_info_.angle);
+
+    const auto range_limit = - (local_range_ + range_) * ec_[0];
+
+    const auto cbCost = [this](
+        const Astar::Vec &s, Astar::Vec &e,
+        const Astar::Vec &v_goal, const Astar::Vec &v_start,
+        const bool hyst) -> float
+    {
+      Astar::Vec d_raw = e - s;
+      d_raw[2] = 0;
+      const Astar::Vec d = d_raw;
+      float cost = euclidCost(d);
+
+      int sum = 0;
+      int num = 0;
+      auto d_index = d;
+      int syaw = s[2];
+      d_index[2] = e[2];
+      d_index.cycleUnsigned(syaw, map_info_.angle);
+      d_index.cycleUnsigned(d_index[2], map_info_.angle);
+      const auto cache_page = motion_interp_cache_[syaw].find(d_index);
```

Comment:
I guess `motion_interp_cache_` should be replaced.

`motion_interp_cache_` has interpolation (arc shaped) between two 3-dof poses.
Since this search is on 2-D space, grid interpolation using `motion_interp_cache_` will not work properly.
In this usage, it can be just a linear interpolation between two 2-D position.